lms/djangoapps/courseware/views/views.py

```diff
@@ -734,6 +735,16 @@ def create_page_context(self, request, course=None, tab=None, **kwargs):
             'openedx.course_experience.reset_course_deadlines', kwargs={'course_id': text_type(course.id)}
         )
 
+        display_reset_dates_banner = False
+
+        if RELATIVE_DATES_FLAG.is_enabled(course.id):
+            course_overview = CourseOverview.objects.filter(id=course.id).first()
+            end_date = getattr(course_overview, 'end_date', None)
+            if not end_date or timezone.now() < end_date:
+                course_enrollment = CourseEnrollment.objects.filter(course=course_overview, user=request.user).first()
+                if course_enrollment and course_enrollment.mode == CourseMode.VERIFIED:
```

Comment:
This could maybe just be `CourseEnrollment.objects.filter(course=course_overview, user=request.user, mode=CourseMode.verified).exists()`, which would prevent you from needing to pull the enrollment object back from the database at all.